fontbe/src/hmtx.rs

```diff
@@ -0,0 +1,80 @@
+//! Generates a [hmtx](https://learn.microsoft.com/en-us/typography/opentype/spec/hmtx) table.
+
+use fontdrasil::orchestration::Work;
+use write_fonts::{
+    dump_table,
+    tables::{hmtx::Hmtx, vmtx::LongMetric},
+};
+
+use crate::{
+    error::Error,
+    orchestration::{BeWork, Context},
+};
+
+pub struct RawHmtx {
+    pub buf: Vec<u8>,
+}
+
+struct HmtxWork {}
+
+pub fn create_hmtx_work() -> Box<BeWork> {
+    Box::new(HmtxWork {})
+}
+
+impl Work<Context, Error> for HmtxWork {
+    /// Generate [hmtx](https://learn.microsoft.com/en-us/typography/opentype/spec/hmtx)
+    fn exec(&self, context: &Context) -> Result<(), Error> {
+        let static_metadata = context.ir.get_final_static_metadata();
+        let var_model = &static_metadata.variation_model;
+        let default_location = var_model.default_location();
+
+        let mut long_metrics: Vec<LongMetric> = static_metadata.glyph_order.iter()
+            .map(|gn| {
+                let bbox = context.get_glyph(gn).bbox();
+                let ir_glyph = context.ir.get_glyph_ir(gn);
+                let Some(ir_instance) = ir_glyph.sources.get(default_location) else {
+                    panic!("{gn} is not defined at the default location. Unnecessary limitation if it covers the default.");
```

Comment:
I would remove that from the error message, as too opinionated: "Unnecessary limitation if it covers the default"

right now, we require the the default location is complete and only allow non-default location to be sparse. and hmtx makes no exception. Changing this assumption is not as trivial as you might think, as it would require to make a temporary VariationModel where another location is assumed to be the default, then the desired default be interpolated from the available ones and re-inserted as the actual default in a new model; but this might also change the effect which the rest of the preexisting masters have, since the default/neutral master by definition applies throughout the variation space, being the base for the other masters..
Anyway that's how OT variation model work nowadays, and font designers know this and don't need to be reminded of this 'limitation' here. File an issue perhas if you like to discuss this further.

Reply:
Fair point. I do think sources that *don't* cover the default shoudl work but it's probably not helpful to note it everywhere, it just needs to be it's own project.